### ISSUE TITLE
Android Crash after taking a shot

### ISSUE THREAD

**Issue by iddan** [gre/react-native-view-shot#118]
### bug report

After capturing a snapshot with ViewShot:

```jsx
       <ScrollView>
               <ViewShot
                    ref={this.handleViewShot}
                    options={{
                        result: 'data-uri',
                        format: 'png',
                    }}
                    snapshotContentContainer
                >
```

```jsx
   share = async () => {
        const { navigation } = this.props
        const title = setTitle(navigation.state.params.chiefComplaintTitle)
        const uri = await this.viewShot.capture()
        const shareImageBase64 = {
            title,
            url: uri,
            subject: title,
        }
        Share.open(shareImageBase64)
    }
```

#### Version & Platform

```bash
react-native-view-shot@2.1.1
```

**Platform:** Android

#### Expected behavior

To take a snapshot (in IOS it works)

#### Actual behavior

![screen shot 2017-11-16 at 15 59 40](https://user-images.githubusercontent.com/12671072/32895028-5c520a74-cae7-11e7-964b-b117a8f272b5.png)

#### Steps to reproduce the behavior
try to use this.viewShot.capture()

**Comment by gre** [comment]
do you have any error message / console message with more information on the error?

**Comment by jackdclark** [comment]
I've also managed to recreate this image on Android 8.0.0 with the following steps

```javascript
componentDidMount () {
  this.share()
}

_onNavigatorEvent (event) {
  if (event.type === 'NavBarButtonPress' && event.id === 'share') {
    return this._share()
  }
}

share = async () => {
  const allowedStorage = await permissions.requestStoragePermission()
  if (allowedStorage) {
    const url = await this.viewShotRef.capture()
    await Share.open({ title: this.props.currentTrip.name, url })
  }
}

render () {
  return (
    <ViewShot ref={ref => (this.viewShotRef = ref)} options={{ result: 'data-uri' }}>
      ...
    </ViewShot>
  )
}
```

The first call in `componentDidMount` works as expected & get the url, all further calls after that cause the error in the PR description above.

I get this in the Android logs

```
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: Exception in native call
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: java.lang.NullPointerException
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at java.io.File.<init>(File.java:282)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at fr.greweb.reactnativeviewshot.RNViewShotModule.releaseCapture(RNViewShotModule.java:58)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at java.lang.reflect.Method.invoke(Native Method)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:363)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:162)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at android.os.Handler.handleCallback(Handler.java:789)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at android.os.Handler.dispatchMessage(Handler.java:98)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at android.os.Looper.loop(Looper.java:164)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:194)
11-16 15:50:19.668 31034 31078 E unknown:ReactNative: 	at java.lang.Thread.run(Thread.java:764)
```

**Comment by gre** [comment]
ok I see, I think there is a bug in data-uri mode and the JS code should never call releaseSnapshot in such mode.

**Comment by khaled-cliqz** [comment]
@iddan I believe this is fixed by that commit https://github.com/gre/react-native-view-shot/commit/153dbbebddd0882cdb2a95b04c8dedfd7292ee2e

**Comment by iddan** [comment]
How can I enjoy it?

**Comment by khaled-cliqz** [comment]
when then next release is made, https://github.com/gre/react-native-view-shot/issues/121

**Comment by iddan** [comment]
cool, I'll keep following

**Comment by khaled-cliqz** [comment]
@iddan a work around is to use `captureRef` function direrctly
```
<ScrollView
      ref="viewshotview"
  <ScrollView>
```
and instead of `await this.viewShotRef.capture()`
you can use `await captureRef(this.viewShotRef, {
                        result: 'data-uri',
                        format: 'png',
                        snapshotContentContainer: true
                    })`

don't forget to `import { captureRef }` instaed of importing `ViewShot` ;)

**Comment by iddan** [comment]
Aren't refs should only be provided as functions?

**Comment by gre** [comment]
going to release asap. ref as string works if your component is a class component (in such case you have the `this.refs`). but not if it's a function component (which kinda make sense because it's a function, there is no `this.refs`).

**Comment by khaled-cliqz** [comment]
I did it for simplification but it works for now if you use it as a string
https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs

**Comment by iddan** [comment]
I am just curious why people use legacy refs everywhere

**Comment by iddan** [comment]
was this resolved?

**Comment by maxkomarychev** [comment]
@iddan [commit](https://github.com/gre/react-native-view-shot/commit/153dbbebddd0882cdb2a95b04c8dedfd7292ee2e) mentioned by @khaled-cliqz is available in most recent release (up to date) - 2.2.0 on npm.

**Comment by gre** [comment]
it should
